File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Core/AlibabaCloud/SDK/ROS/CDK/Core/IConstruct.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Core
{
    /// <summary>Represents a construct.</summary>
    [JsiiInterface(nativeType: typeof(IConstruct), fullyQualifiedName: "@alicloud/ros-cdk-core.IConstruct")]
    public interface IConstruct : Constructs.IConstruct, AlibabaCloud.SDK.ROS.CDK.Core.IDependable
    {
        /// <summary>The construct tree node for this construct.</summary>
        [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
        AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
        {
            get;
        }

        /// <summary>Represents a construct.</summary>
        [JsiiTypeProxy(nativeType: typeof(IConstruct), fullyQualifiedName: "@alicloud/ros-cdk-core.IConstruct")]
        new internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Core.IConstruct
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkcore/IConstruct.go
package alicloudroscdkcore

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore/internal"
	"github.com/aws/constructs-go/constructs/v3"
)

// Represents a construct.
type IConstruct interface {
	constructs.IConstruct
	IDependable
	// The construct tree node for this construct.
	Node() ConstructNode
}

// The jsii proxy for IConstruct
type jsiiProxy_IConstruct struct {
	internal.Type__constructsIConstruct
	jsiiProxy_IDependable
}

func (j *jsiiProxy_IConstruct) Node() ConstructNode {
	var returns ConstructNode
	_jsii_.Get(
		j,
		"node",
		&returns,
	)
	return returns
}

